polkadot-parachain/src/service.rs

```diff
@@ -375,6 +375,16 @@ where
 	let prometheus_registry = parachain_config.prometheus_registry().cloned();
 	let transaction_pool = params.transaction_pool.clone();
 	let import_queue = cumulus_client_service::SharedImportQueue::new(params.import_queue);
+	let warp_sync_params =
+		match cumulus_client_network::WaitForParachainTargetBlock::<Block>::warp_sync_get(
+			para_id,
+			relay_chain_interface.clone(),
+		)
+		.await
```

Comment:
You also don't want to `await` here. This should be some background "process" that waits for the target block and then forwards it to sync using the channel.